File: CardMasterSkin/Elements/SECircle.vb
Imports System.Drawing.Drawing2D
Imports CardMasterCard.Card
Imports CardMasterSkin.GraphicsElement

Namespace Skins

    Public Class SECircle
        Inherits SkinElement

        Dim m_radius As Integer

        Public Sub New(skin As Skin, radius As Integer)
            Me.New(skin, 0, 0, radius * 2, radius * 2, radius)
        End Sub

        Public Sub New(skin As Skin, x As Integer, y As Integer, width As Integer, height As Integer, radius As Integer)
            MyBase.New(skin, x, y, width, height)

            m_radius = radius

        End Sub

        Protected Overrides Function GetGraphicElements(card As Card) As List(Of GraphicElement)

            Dim graphicElementsList As New List(Of GraphicElement)()
            Dim path As New GraphicsPath()

            path.AddEllipse(X, Y, Width, Height)

            graphicElementsList.Add(New PathElement(path, GetBackground))

            path = Nothing

            Return graphicElementsList

        End Function


    End Class

End Namespace

File: CardMasterSkin/Elements/SECircle.cs
using CardMasterCard.Card;
using CardMasterSkin.GraphicsElements;
using CardMasterSkin.Skins;
using System.Collections.Generic;
using System.Drawing.Drawing2D;

namespace CardMasterSkin.Elements
{
    public class SECircle : SkinElement
    {
        public SECircle(Skin skin, int x, int y, int radius) : base(skin, x, y, radius * 2, radius * 2)
        { }

        public SECircle(Skin skin, int x, int y, int width, int height) : base(skin, x, y, width, height)
        { }

        protected override List<GraphicElement> GetGraphicElements(Card card)
        {
            var graphicElementsList = new List<GraphicElement>();
            var path = new GraphicsPath();

            path.AddEllipse(X, Y, Width, Height);

            graphicElementsList.Add(new PathElement(path, GetBackground()));
            
            path = null;

            return graphicElementsList;
        }
    }
}
